cni/cmd/tc-redirect-tap/main.go

```diff
@@ -376,3 +398,18 @@ type NoPreviousResultError struct{}
 func (e NoPreviousResultError) Error() string {
 	return "no previous result was found, was this plugin chained with a previous one?"
 }
+
+// extractArgs returns cli args in form of map of strings
+// args string - cli args string("key1=val1;key2=val2)
+func extractArgs(args string) map[string]string {
+	result := make(map[string]string)
+	argumentsPairs := strings.Split(args, ";")
+	for _, pairStr := range argumentsPairs {
+		pair := strings.Split(pairStr, "=")
+		if len(pair) > 1 {
+			result[pair[0]] = pair[1]
+		}
```

Comment:
Just some very minor additional corner cases to handle:
* If `extractArgs` was provided `"KEY=foo=blah"`, then we'd get a map where `KEY` is set to `foo` and the rest of the arg val is ignored. I just tried, it apparently is allowed to use an `=` character in a tap device name, so this is something we should handle correctly even though it's extremely obscure.
* If we were provided something like `"TC_REDIRECT_TAP_UID;TC_REDIRECT_TAP_GID"`, we'd just ignore the args and use defaults, which is a bit surprising for users because `"TC_REDIRECT_TAP_UID=;TC_REDIRECT_TAP_GID="` would result in an error.

I think both of these issues can be addressed by using `strings.SplitN(pairStr, "=", 2)` and then returning an error from this function if `len(pair) != 2` (just saying that the arg string is in an invalid format).